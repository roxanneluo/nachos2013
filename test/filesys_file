Purpose:
Test file with increasing size

Usage:
-- nachos.ag.FilesysGrader -x filesys_dir.coff -# testRoot=XXX,
coffPar0=XXX,coffPar1=XXX,output=filesys_dir.out

Author:
Bo Tang

History:
2010-01-06 Bo Tang: The first version.

Arguments:
-output filesys_file.out
-coffPar0 the number of bytes you add to the file for each time.
-coffPar1 the number of time you add to the file. 
MAKE SURE:  coffPar0 * coffPar1 less than your disk MAXFILESIZE(maybe 2MB).
And coffPar1 <= 100, or you will mismactch the ouput.

Restrictions:
To run this test, you should follow the specification in filesys_spec, and 
support the testSysCall getFreeDiskSize() which CONTAINS the sectors of your
SWAPFILE takes.

Sample:
-- nachos.ag.FilesysGrader -x filesys_file.coff -[] ./conf/proj5.conf -# 
testRoot=../nachos-test/,output=filesys_file.out,coffPar0=10240,coffPar1=100

Note:
You will need more  stack pages.