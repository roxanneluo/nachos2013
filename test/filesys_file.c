#include "filesysgrader.h"
#include "stdio.h"
#include "stdlib.h"

char buf[1024*1024*2];

int main(int argc, char** argv)
{
    int size = getFreeDiskSize();
    int count = readParameter(0);
    int times = readParameter(1);
    int fp  = creat("file1");
    assertTrueWMsg(fp >= 0, "creat failed");
    int i;
    for (i = 0; i < times; i++){
        assertTrueWMsg(write(fp, buf, count) == count, "write failed");
        printf("create %d\r\n", i);
    }
    close(fp);
    assertTrueWMsg(unlink("file1") == 0, "unlink failed");
    assertTrueWMsg(size == getFreeDiskSize(), "size mismatch");
    done();
    return 0;
}
