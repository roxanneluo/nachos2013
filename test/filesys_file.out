create 0
create 1
create 2
create 3
create 4
create 5
create 6
create 7
create 8
create 9
create 10
create 11
create 12
create 13
create 14
create 15
create 16
create 17
create 18
create 19
create 20
create 21
create 22
create 23
create 24
create 25
create 26
create 27
create 28
create 29
create 30
create 31
create 32
create 33
create 34
create 35
create 36
create 37
create 38
create 39
create 40
create 41
create 42
create 43
create 44
create 45
create 46
create 47
create 48
create 49
create 50
create 51
create 52
create 53
create 54
create 55
create 56
create 57
create 58
create 59
create 60
create 61
create 62
create 63
create 64
create 65
create 66
create 67
create 68
create 69
create 70
create 71
create 72
create 73
create 74
create 75
create 76
create 77
create 78
create 79
create 80
create 81
create 82
create 83
create 84
create 85
create 86
create 87
create 88
create 89
create 90
create 91
create 92
create 93
create 94
create 95
create 96
create 97
create 98
create 99

